flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -1596,6 +1608,12 @@ CompletableFuture<Void> getJobTerminationFuture(JobID jobId) {
                 jobId, CompletableFuture.completedFuture(null));
     }
 
+    @VisibleForTesting
+    CompletableFuture<Void> addJobTerminationFuture(
```

Comment:
We could reuse the method registerJobManagerRunnerTerminationFuture and annotate it as VisibleForTesting.